trestle/tasks/osco_to_oscal.py

```diff
@@ -0,0 +1,184 @@
+                    parts = ifile.parts
+                    ifn = parts[len(parts)-1]
+                    if ifn.endswith('oscal-metadata.yaml'):
+                        continue
+                    if ifn.endswith('oscal-metadata.yml'):
+                        continue
+                    ofile = self._calculate_ofile(ifn, odir)
+                    if not overwrite:
+                        if os.path.exists(ofile):
+                            logger.error(f'file exists: {ofile}')
+                            return TaskOutcome('simulated-failure')
+                    mfile = idir / 'oscal-metadata.yaml'
+                    metadata = self._get_metadata(mfile)
+                    logger.debug(f'create: {ofile}')
+                    idata = self._read_content(ifile)
+                    odata, analysis = osco.get_observations(idata, metadata)
+                    self._write_content(ofile, odata, True)
+                    logger.debug(f'Rules Analysis:')
+                    logger.debug(f'config_maps: {analysis["config_maps"]}')
+                    logger.debug(f'dispatched rules: {analysis["dispatched_rules"]}')
+                    logger.debug(f'result types: {analysis["result_types"]}')
+                return TaskOutcome('simulated-success')
+            logger.error(f'config missing')
+            return TaskOutcome('simulated-failure')
+        except Exception:
+            traceback.print_exc()
+            return TaskOutcome('simulated-failure')
+
+    def execute(self) -> TaskOutcome:
+        """Provide an actual outcome."""
+        try:
+            if self._config:
+                idir = pathlib.Path(self._config.get('input-dir'))
+                if idir is None:
+                    logger.error(f'config missing "input-dir"')
+                    return TaskOutcome('failure')
+                odir = pathlib.Path(self._config.get('output-dir'))
+                if odir is None:
+                    logger.error(f'config missing "output-dir"')
+                    return TaskOutcome('failure')
+                overwrite = self._config.getboolean('output-overwrite', True)
+                quiet = self._config.getboolean('quiet', False)
+                os.makedirs(odir, exist_ok = True) 
+                for ifile in sorted(pathlib.Path(idir).iterdir()):
+                    parts = ifile.parts
+                    ifn = parts[len(parts)-1]
+                    if ifn.endswith('oscal-metadata.yaml'):
+                        continue
+                    if ifn.endswith('oscal-metadata.yml'):
+                        continue
+                    ofile = self._calculate_ofile(ifn, odir)
+                    if not overwrite:
+                        if os.path.exists(ofile):
+                            logger.error(f'file exists: {ofile}')
+                            return TaskOutcome('failure')
+                    mfile = idir / 'oscal-metadata.yaml'
```

Comment:
This line is inconsistent with the assumptions in line 123

Reply:
Not sure what you mean here by inconsistent? This is checking if the output file already exists and if so overwriting only if permitted. Previous lines (e.g., 123) are concerned with input, I believe, and and therefore not germane.

Reply:
Okay reading it again - It looks to me like 122 and mfile point to the same location (unless I am wrong) which would imply that the metadata will never get read if it exists. Unless I am missing something.

Reply:
Not sure what is inconsistent? There are 3 kinds of files: 1) input osco yaml 2) input oscal metadata 3) output oscal. 

1) are the input files to be converted from OSCO formatted yaml into OSCAL formatted json
2) is optional an yaml file that, if present, augments the produced OSCAL formatted json
3) are output json files, one for each of the input yaml files in 1)

If should be noted that if the output file to be produced already exists, then it will not be replaced unless the config specifies overwrite = True, which is the default. That is specify overwrite = False in the config to prevent overwriting.

Reply:
My claim is that updated code and comments now address your "inconsistent with the assumptions" concern.